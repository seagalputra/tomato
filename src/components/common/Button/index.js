import React from 'react'
import PropTypes from 'prop-types'

const Button = ({ children, variant, isDisabled }) => {
  if (isDisabled)
    return (
      <button
        type="button"
        disabled
        className="bg-blue-500 opacity-75 text-white font-bold py-2 px-4 rounded"
      >
        {children}
      </button>
    )

  return (
    <button
      type="button"
      className={`${
        variant === 'primary'
          ? 'bg-blue-500 hover:bg-blue-700 text-white font-bold'
          : 'bg-transparent hover:bg-blue-500 text-blue-700 hover:text-white border border-blue-500 hover:border-transparent font-semibold'
      } py-2 px-4 rounded`}
    >
      {children}
    </button>
  )
}

Button.propTypes = {
  children: PropTypes.string.isRequired,
  variant: PropTypes.string,
  isDisabled: PropTypes.bool
}

Button.defaultProps = {
  variant: 'primary',
  isDisabled: false
}

export default Button
