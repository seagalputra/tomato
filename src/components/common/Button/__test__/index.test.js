import React from 'react'
import { shallow } from 'enzyme'
import Button from '..'

describe('Button', () => {
  it('renders without crashing', () => {
    const component = shallow(<Button>This is Button</Button>)
    expect(component).toMatchSnapshot()
  })
})
