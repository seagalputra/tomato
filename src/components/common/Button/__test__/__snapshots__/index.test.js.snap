// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders without crashing 1`] = `
<button
  className="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded"
  type="button"
>
  This is Button
</button>
`;
